.github/workflows/esrupgrade.yml

```diff
@@ -0,0 +1,92 @@
+      - name: Pull EE image
+        run: |
+          docker pull $BUILD_IMAGE
+      - name: Run migration through server
+        run: |
+          mkdir -p client/plugins
+          cd build
+          # Run the server in the background to trigger the migrations
+          docker run --name mmserver \
+            --net ghactions_mm-test \
+            --ulimit nofile=8096:8096 \
+            --env-file=dotenv/test.env \
+            --env MM_SQLSETTINGS_DRIVERNAME="mysql" \
+            --env MM_SQLSETTINGS_DATASOURCE="mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8&multiStatements=true" \
+            -v ~/work/mattermost-server:/mattermost-server \
+            -w /mattermost-server/mattermost-server \
+            $BUILD_IMAGE &
+          # In parallel, wait for the migrations to finish.
+          # To verify this, we check whether the db_migrations table contains the latest migration in this version
+          CHECK_LATEST_MIGRATION_CMD="SELECT COUNT(1) FROM db_migrations WHERE Version = $LATEST_MIGRATION;"
+          until [ `docker exec ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS -e "$CHECK_LATEST_MIGRATION_CMD"` -eq 1 ]; do\
+            echo "Waiting for migrations to finish..."; \
+            sleep 1; \
+            done;
+          # Make sure to stop the server. Also, redirect output to null;
+          # otherwise, the name of the container gets written to the console, which is weird
+          docker stop mmserver > /dev/null
+      - name: Dump upgraded database
+        run: |
+          docker exec -i ghactions_mysql_1 mysqldump --skip-opt --set-gtid-purged=OFF $MYSQL_CONN_ARGS > dump.7.8.server.sql
+      - name: Reset DB
+        run: |
+          # Build a script to drop all tables in the DB ignoring any foreign key checks
+          docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS -e "SELECT concat('DROP TABLE IF EXISTS \`', table_name, '\`;') FROM information_schema.tables WHERE table_schema = 'mattermost_test';" > drop-queries.sql
+          echo "SET FOREIGN_KEY_CHECKS = 0;" > drop-script.sql
+          cat drop-queries.sql >> drop-script.sql
+          echo "SET FOREIGN_KEY_CHECKS = 1;" >> drop-script.sql
+          # Drop all tables in the DB
+          docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS < drop-script.sql
+      - name: Initialize the database with the source DB dump
+        run: |
+          curl $DB_DUMP_URL | zcat | docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS
+      - name: Run migration through script
+        run : |
+          # Build the image with the migration wrapper
+          cd scripts/esrupgrades
+          docker build -t esrupgrade .
+          # Run the image with the expected driver, DSN, and migration file
+          docker run -i --rm --net ghactions_mm-test esrupgrade mysql "mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8&multiStatements=true" esr.5.37-7.8.mysql.up.sql
```

Comment:
Out of curiosity, can we just use the `mysql` tooling to ingest this file directly? What does the Go tooling offer?

Reply:
Yes, the answer is what Ibrahim pointed out.

Your proposed command fails both locally and in CI with:
```
ERROR 1064 (42000) at line 25: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 17
```

Check this commit in the playground branch I have for testing: https://github.com/mattermost/mattermost-server/commit/6a96473216831720bc49459c02f4860c76834f23

Reply:
Hmm, if we can't just do `mysql < script.sql`, how will our customers leverage this script?

Are we needing to do something like what's described in https://dba.stackexchange.com/questions/41336/how-to-create-a-mysql-stored-procedure-through-linux-terminal?

Reply:
That's a good point. I don't know how the customer that triggered all this effort actually runs their SQL commands. Let me ask, and depending on that we could test the `DELIMITER` stuff.

Reply:
Interestingly I can run this command locally with our sql files :/ I don't know if I configured something else differently.

Reply:
@isacikgoz, have you tried with the script in this PR? I've found the issue only with specific individual migrations, as well as with this script (understandable, since it contains such migrations).

Reply:
Ah okay, that's not related to multiStatements 🤦 It's about the quotes, there should be a fix for that. Can you look into that if there is a fix? ref: https://stackoverflow.com/questions/75598918/stored-procedure-with-single-quotes

Reply:
There are two hard problems at computer science and all three of them are wrapped in different quotes.

Reply:
Now seriously, I'll try to see if this is the real cause and find a solution.

Reply:
So yes, the problem was indeed the lack of `DELIMITER` statements, not anything quote-related. The documentation is actually pretty clear on this topic: https://dev.mysql.com/doc/refman/8.0/en/stored-programs-defining.html
Please see https://github.com/mattermost/mattermost-server/pull/22448/commits/fac26e2eadeba019199866312f56d019c87f6cfd for a very comforting commit.